subprojects/gradle/test-summary/src/main/kotlin/com/avito/test/summary/model/CrossDeviceSuite.kt

```diff
@@ -25,25 +26,34 @@ public data class CrossDeviceSuite(val crossDeviceRuns: List<CrossDeviceRunTest>
     val lostOnSomeDevicesCount: Int
         get() = consistentData.count { it.status is CrossDeviceStatus.LostOnSomeDevices }
 
-    val percentLostOnSomeDevicesOfAutomated: Percent
-        get() = lostOnSomeDevicesCount.percentOf(automatedCount)
-
-    val inconsistentCount: Int
-        get() = crossDeviceRuns.count { it.status is CrossDeviceStatus.Inconsistent }
-
-    val percentSuccessOfAutomated: Percent = success.percentOf(automatedCount)
-
-    val percentSkippedOnAllDevicesOfAutomated: Percent = skippedOnAllDevicesCount.percentOf(automatedCount)
-
     val failedOnAllDevicesCount: Int
         get() = consistentData.count { it.status is CrossDeviceStatus.FailedOnAllDevices }
 
-    val percentFailedOnAllDevicesOfAutomated: Percent = failedOnAllDevicesCount.percentOf(automatedCount)
-
     val failedOnSomeDevicesCount: Int
         get() = consistentData.count { it.status is CrossDeviceStatus.FailedOnSomeDevices }
 
-    val percentFailedOnSomeDevicesOfAutomated: Percent = failedOnSomeDevicesCount.percentOf(automatedCount)
+    val percentLostOnSomeDevicesOfAutomated: Percent
+    val percentSuccessOfAutomated: Percent
+    val percentSkippedOnAllDevicesOfAutomated: Percent
+    val percentFailedOnAllDevicesOfAutomated: Percent
+    val percentFailedOnSomeDevicesOfAutomated: Percent
+
+    init {
+        if (automatedCount == 0) {
+            val zeroPercents = 0F.fromZeroToOnePercent()
+            percentSuccessOfAutomated = zeroPercents
```

Comment:
Can `percentSuccessOfAutomated` and other `percent...` have non-zero values in this case?
If not it may be good to check this precondition explicitly.
Otherwise, it's non-obvious why N is 0% from 0.

Reply:
As I understand it's normal case when you have zero automated tests and two manual tests.

But it's incorrect to count manual to automated. Better is count manual comaparing with total. So, I could rewrite percents logic